app/fakefish.py

```diff
@@ -172,21 +164,36 @@ def set_env_vars(bmc_endpoint, username, password):
     my_env["BMC_PASSWORD"] = password
     return my_env
 
-def run():
+def run(port, debug, tls_mode):
     """
 
     """
-    app.run(host='0.0.0.0', port=port, debug=debug, ssl_context='adhoc')
+    if tls_mode == 'adhoc':
+        app.run(host='0.0.0.0', port=port, debug=debug, ssl_context='adhoc')
```

Comment:
Side comment: '0.0.0.0' is not IPv6-friendly, I assume. Maybe `::`

Reply:
Done.